examples/access_traits/example_cuda_access_traits.cpp

```diff
@@ -77,21 +77,25 @@ int main(int argc, char *argv[])
   cudaStreamCreate(&stream);
   cudaMemcpyAsync(d_a, a.data(), sizeof(a), cudaMemcpyHostToDevice, stream);
 
-  Kokkos::Cuda cuda{stream};
-  ArborX::BVH<Kokkos::CudaSpace> bvh{cuda, PointCloud{d_a, d_a, d_a, N}};
-
-  Kokkos::View<int *, Kokkos::CudaSpace> indices("Example::indices", 0);
-  Kokkos::View<int *, Kokkos::CudaSpace> offset("Example::offset", 0);
-  ArborX::query(bvh, cuda, Spheres{d_a, d_a, d_a, d_a, N}, indices, offset);
-
-  Kokkos::parallel_for(
-      "Example::print_indices", Kokkos::RangePolicy<Kokkos::Cuda>(cuda, 0, N),
-      KOKKOS_LAMBDA(int i) {
-        for (int j = offset(i); j < offset(i + 1); ++j)
-        {
-          printf("%i %i\n", i, indices(j));
-        }
-      });
+  {
+    // Place the code using CUDA instance in a block to make sure the instance
+    // is destroyed prior to destroying the stream.
```

Comment:
Would it guarantee the destroy order with the instance, though?

Reply:
The instance would get out of scope and destroyed first, then the kokkos execution environment scope guard destructor kicks in, calls `Kokkos::finalize` which invokes all registered callbacks before releasing resources.

Reply:
That is "yes it would do the right thing".
I prefer that solution but I am not blocking.

Reply:
Alternatively, you could just move `Kokkos::ScopeGuard guard(argc, argv);` inside this block so that the stream is created/destroyed before/after Kokkos is initialized/finalized.

Reply:
I don't think it matters whether the stream is destroyed before/after Kokkos is finalized. As long as the instance is destroyed before, that's the only thing that matters, if I understand it right. So, I would prefer to keep the current code.

Reply:
The suggestion I suggest is more explicit about what needs to happen, hence more readable IMO.